jwt_proxy/api.py

```diff
@@ -2,12 +2,41 @@
 import jwt
 import requests
 import json
+import re
 
 from jwt_proxy.audit import audit_HAPI_change
 
 blueprint = Blueprint('auth', __name__)
 SUPPORTED_METHODS = ('GET', 'POST', 'PUT', 'DELETE', 'OPTIONS')
 
+# TODO: to be pulled into its own module and loaded per config
+def scope_filter(req, token):
+    # Check path
+    resource_pattern = rf"(Patient|DocumentReference)$"
```

Comment:
I fear this approach leaves a backdoor to well tailored requests.  A user could include a string like "Patient" in say an or-clause of a search, or the like.

Any reason to not just limit to the requested resource in the request path?  i.e. `{url}/Patient` or `{url}/DocumentReference` for this use case?

Reply:
> Any reason to not just limit to the requested resource in the request path? i.e. {url}/Patient or {url}/DocumentReference for this use case?
That's the intention of this pattern, to enforce the request path (sans params) ends in either `Patient` or `DocumentReference` (it's checked against `req.path`, which will typically be `fhir/DocumentReference` etc. I preferred containing this check within the function to defining a url rule or something, but let me know if there's a better/more pythonic way!